### DISCUSSION TITLE
how to set session timeout for mysql connection

### DISCUSSION THREAD

**Discussion by Yifan122** [trinodb/trino#15502] [General]
Hi,

I encountered some query failed due to timeout when query mysql.

Could I set the session timeout for the mysql catalog?

**Comment by hashhar** [comment]
Depends a lot on what you mean by "session".

There are connection properties called `connectTimeout` and `socketTimeout` which controls timeouts on the underlying socket. By default there is no timeout.

There is also a server level configuration which controls how long queries can execute called `MAX_EXECUTION_TIME`. You can change it by passing it in connection URL as `?sessionVariables=MAX_EXECUTION_TIME=<value>`.

There is also `interactive_timeout` and `wait_timeout` but those are server level variables.

Can you share what error you actually see? That would help find out which of these needs to be adjusted.
